tests/test_jit.py

```diff
@@ -30,133 +35,44 @@ def test_filter_jit1(api_version, getkey):
         eqx.nn.MLP(2, 2, 2, 2, key=getkey()),
     ]
     array_tree = [{"a": a, "b": b}, (c,)]
-    _mlp = jtu.tree_map(lambda u: u if eqx.is_array_like(u) else None, general_tree[-1])
-
-    if api_version == 0:
-
-        @ft.partial(eqx.filter_jit, filter_spec=lambda _: True)
-        def f(x):
-            return x
-
-    else:
-
-        @eqx.filter_jit(default=True)
-        def f(x):
-            return x
-
-    assert jnp.all(a == f(a))
-    f1 = f(array_tree)
-    assert jnp.all(f1[0]["a"] == a)
-    assert jnp.all(f1[0]["b"] == b)
-    assert jnp.all(f1[1][0] == c)
-
-    with pytest.raises(TypeError):
-        f(general_tree)
-
-    def g(x):
-        return jtu.tree_map(lambda u: u if eqx.is_array_like(u) else None, x)
-
-    if api_version == 0:
-        g = eqx.filter_jit(g, filter_spec=eqx.is_inexact_array)
-    else:
-        g = eqx.filter_jit(default=eqx.is_inexact_array)(g)
-
-    assert jnp.all(a == g(a))
-    g1 = g(array_tree)
-    assert jnp.all(g1[0]["a"] == a)
-    assert jnp.all(g1[0]["b"] == b)
-    assert jnp.all(g1[1][0] == c)
-    g2 = g(general_tree)
-    assert _eq(g2[0], 1)
-    assert _eq(g2[1], True)
-    assert _eq(g2[2], None)
-    assert jnp.all(g2[3]["a"] == a)
-    assert _eq(g2[3]["tuple"][0], 2.0)
-    assert jnp.all(g2[3]["tuple"][1] == b)
-    assert jnp.all(g2[4] == c)
-    assert _eq(g2[5], _mlp)
-
-    def h(x):
-        return jtu.tree_map(lambda u: u if eqx.is_array_like(u) else None, x)
-
-    if api_version == 0:
-        h = eqx.filter_jit(h, filter_spec=eqx.is_array_like)
-    else:
-        h = eqx.filter_jit(h, default=eqx.is_array_like)
-
-    assert jnp.all(a == h(a))
-    h1 = h(array_tree)
-    assert jnp.all(h1[0]["a"] == a)
-    assert jnp.all(h1[0]["b"] == b)
-    assert jnp.all(h1[1][0] == c)
-    h2 = h(general_tree)
-    assert _eq(h2[0], jnp.array(1))
-    assert _eq(h2[1], jnp.array(True))
-    assert _eq(h2[2], None)
-    assert jnp.all(h2[3]["a"] == a)
-    assert _eq(h2[3]["tuple"][0], jnp.array(2.0))
-    assert jnp.all(h2[3]["tuple"][1] == b)
-    assert jnp.all(h2[4] == c)
-    assert _eq(h2[5], _mlp)
-
-
-@pytest.mark.parametrize("api_version", (0, 1))
-def test_filter_jit2(api_version, getkey):
-    a = jrandom.normal(getkey(), (2, 3))
-    b = jrandom.normal(getkey(), (3,))
-    c = jrandom.normal(getkey(), (1, 4))
-    general_tree = [
-        1,
-        True,
-        object(),
-        {"a": a, "tuple": (2.0, b)},
-        c,
-        eqx.nn.MLP(2, 2, 2, 2, key=getkey()),
-    ]
-    _mlp = jtu.tree_map(lambda u: u if eqx.is_array_like(u) else None, general_tree[-1])
+    mlp_add = jtu.tree_map(lambda u: u + 1 if eqx.is_array(u) else u, general_tree[-1])
 
-    spec = [
-        True,
-        True,
-        False,
-        {"a": True, "tuple": (False, True)},
-        True,
-        eqx.is_inexact_array,
-    ]
+    array_tree_ = jax.tree_map(
+        lambda x: jnp.copy(x) if eqx.is_array(x) else x, array_tree
+    )
```

Comment:
Nit: `jtu.tree_map(jnp.copy, array_tree)` suffices as all elements are arrays.
Also note the use of `jax.tree_util` (`jtu`) rather than `jax`. (The `tree_map` utility is being moved.)